### DISCUSSION TITLE
Is there a way to disable a button while the form is submitting with any property of react hook form?

### DISCUSSION THREAD

**Discussion by Jsramosra** [react-hook-form/react-hook-form#1953] [Q&A] [answered]
Actually, it's a pretty simple question, but still haven't found an appropiate solution.
I was trying to trigger the onsubmit boolean on the disabled property of the button, but still doesn't work, Any idea?

**Comment by bluebill1049** [comment, marked as answer]
<img width="764" alt="Screen Shot 2020-06-25 at 8 11 42 am" src="https://user-images.githubusercontent.com/10513364/85632892-82ee0d00-b6bb-11ea-9829-4c378e978177.png">

**Comment by Jsramosra** [comment]
Is it necessary to handle the component as a function?, i'm trying to use isSubmitting but it never changes

```
const MockComponent = props =>{
const {formState, handleSubmit} = useForm();

const onSubmit = values => console.log("is submitting");

return(
<form onsubmit={handleSubmit(onSubmit)}>
  //Material-ui button
<Button type="submit" id="submitButton" disabled={formState.isSubmitting} variant="contained" color="secondary">
           Send
          </Button>

</form>
);

}
```

**Reply by bluebill1049** [reply]
make sure you read the value before `render`

`const { isSubmitting } = formState`

<img width="966" alt="Screen Shot 2020-06-25 at 1 33 23 pm" src="https://user-images.githubusercontent.com/10513364/85650243-72a05700-b6e8-11ea-93e3-02b3eddf57e4.png">
